pkg/js/validator/validate-store.ts

```diff
@@ -0,0 +1,976 @@
+    // Collect valid tuples
+    const tuples = [];
+    if (store.tuples && store.tuples.length) {
+      tuples.push(...store.tuples);
+    }
+
+    const test = store.tests[testNo];
+    if (!test) {
+      continue;
+    }
+
+    if (test.tuples && test.tuples.length) {
+      tuples.push(...test.tuples);
+    }
+
+    // Validate check
+    for (const checkNo in test.check) {
+      if (!test.check[checkNo] || !test.check[checkNo].user || !test.check[checkNo].object) {
+        return false;
+      }
+      errors.push(
+        ...validateCheck(
+          model,
+          test.check[checkNo],
+          tuples,
+          params,
+          instancePath + `/tests/${testNo}/check/${checkNo}`,
+        ),
+      );
+    }
+
+    // Validate list objects
+    for (const listNo in test.list_objects) {
+      if (!test.list_objects[listNo] || !test.list_objects[listNo].user || !test.list_objects[listNo].type) {
+        return false;
+      }
+      errors.push(
+        ...validateListObject(
+          model,
+          test.list_objects[listNo],
+          tuples,
+          params,
+          instancePath + `/tests/${testNo}/list_objects/${listNo}`,
+        ),
+      );
+    }
+
+    for (const listNo in test.list_users) {
+      if (!test.list_users[listNo] || !test.list_users[listNo].object || !test.list_users[listNo].user_filter) {
+        return false;
+      }
+      errors.push(
+        ...validateListUsers(
+          model,
+          test.list_users[listNo],
+          tuples,
+          params,
+          instancePath + `/tests/${testNo}/list_users/${listNo}`,
+        ),
+      );
+    }
+  }
+
+  if (errors.length) {
+    validateStore.errors?.push(...errors);
+    return false;
+  }
+  return true;
+}
+
+const validateStore: SchemaValidateFunction = function (
+  this: { jsonModel: AuthorizationModel },
+  store: Store,
+  cxt: { instancePath: string },
+): boolean {
+  validateStore.errors = validateStore.errors || [];
+
+  // Require model or model_file
+  if (!store.model && !store.model_file) {
+    validateStore.errors.push(modelOrModelFile(cxt.instancePath));
+    return false;
+  }
+
+  return validateTestTypes(store, this.jsonModel, cxt.instancePath);
+};
+
+// YAML validation using ajv
+export function YamlTuplesValidator(): ValidateFunction {
+  return new Ajv({
+    allErrors: true,
+    verbose: true,
+    passContext: true,
+    $data: true,
+  })
+    .addFormat("user", {
+      validate: Validate.user,
+    })
+    .addFormat("relation", {
+      validate: Validate.relation,
+    })
+    .addFormat("object", {
+      validate: Validate.object,
+    })
+    .addFormat("condition", {
+      validate: Validate.relationshipCondition,
+    })
+    .addKeyword({
+      keyword: "valid_tuple",
+      type: "object",
+      schema: false,
+      errors: true,
+      validate: validateTuple,
+    })
+    .compile(OPENFGA_TUPLES_SCHEMA);
+}
+
+// YAML tuple validation
+const OPENFGA_TUPLES_SCHEMA: Schema = {
```

Comment:
You might want to have this in a separate file if you want to re-use this across go, js and Java. But you can move it out in a separate PR

Reply:
Isn't this just using [JSON Schema](https://json-schema.org/)? That's a standard so should be applicable for Go and Java too.

See https://json-schema.org/implementations

E.g.
Java: https://github.com/networknt/json-schema-validator
Go: https://github.com/kaptinlin/jsonschema